src/Runtime/OMExternalConstant.inc

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---- OMExternalParam.cpp - OMExternalParam C/C++ Implementation ------===//
+//
+// Copyright 2023 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains C/C++ neutral implementation of OMExternalParam.
+//
+//===----------------------------------------------------------------------===//
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+const int i = 1;
+#define IS_SYSTEM_LE() (!((*(const char *)&i) == 0))
+
+#define XOR(a, b) (!(a) != !(b))
+
+void checkEndianness(const char constPackIsLE) {
+  if (XOR(IS_SYSTEM_LE(), constPackIsLE)) {
+    fprintf(stderr, "Constant pack is stored in a byte order that is not "
+                    "native to this current system.");
+    exit(1);
+  }
+}
+
+/// malloc_aligned and free_aligned are adopted from
+/// https://github.com/IBM/zDNN/blob/main/zdnn/malloc4k.c
+
+/// malloc() that does alignment
+///
+/// \param[in] size Size to be malloc'd
+/// \param[in] align Alignment, -1 if no alignment
+///
+/// \return Pointer to the malloc'd area if successful, or NULL otherwise
+///
+/// TODO: find a new home for this function.
+void *omMallocAligned(int64_t size, int64_t align) {
+  // No alignment.
+  if (align == -1) {
+    void *ptr = malloc(size);
+    if (!ptr) {
+      perror("Error during malloc");
+      fprintf(stderr, "errno = %d\n", errno);
+    }
+    return ptr;
+  }
+
+  // Request one more page + size of a pointer from the OS.
+  unsigned short extra_allocation = (align - 1) + sizeof(void *);
+
+  void *ptr = malloc(size + extra_allocation);
+  if (!ptr) {
+    perror("Error during malloc");
+    fprintf(stderr, "errno = %d\n", errno);
+    return ptr;
+  }
+
+  // Find the align boundary after ptr.
+  void *aligned_ptr =
+      (void *)(((uintptr_t)ptr + extra_allocation) & ~(align - 1));
+  // Put the original malloc'd address right before aligned_ptr.
+  ((void **)aligned_ptr)[-1] = ptr;
+
+  return aligned_ptr;
+}
+
+/// free() what was allocated via omMallocAligned()
+///
+/// \param[in] ptr Pointer returned by omMallocAligned()
+///
+/// \return None
+///
+/// TODO: find a new home for this function.
+void omFreeAligned(void *ptr, int64_t align) {
+  if (ptr) {
+    if (align == -1) {
+      // No alignment.
+      free(ptr);
+    } else {
+      // Get the original malloc'd address from where we put it and free it.
+      void *original_ptr = ((void **)ptr)[-1];
+      free(original_ptr);
+    }
+  }
+}
+
```

Comment:
There are already aligned memory allocation functions: `posix_memalign`, on windows it's `_aligned_malloc`.

Reply:
Use them now. Thanks for the suggestion!